Add Legend to Execution Plan

ThinkingBIGRon opened the pull request:
This improves the display of the execution plan by adding a legend to the top right of the page.

Context: @chrisdmacrae presented Nuke to a local developer group, including the execution plan and the group saw the value in the execution plan - they did not immediately understand it.  This PR adds some context to the page.

I confirm that the pull-request:

- [x] Follows the contribution guidelines
- [x] Is based on my own work
- [x] Is in compliance with my employer

ron-myers committed:
Add Legend to Execution Plan

ThinkingBIGRon commented:
Screenshot of this on an example build.  

![OnPaste 20200614-155755](https://user-images.githubusercontent.com/10678906/84603667-74efef00-ae66-11ea-8afe-9b7059c6c9e8.png)

The content on this legend is static.

matkoch commented:
Good idea! But I have a couple of suggestions for improvement:

- the bars above "Will not execute" and "Will execute" could actually show a target rectangle
- show legend in upper left
- arrows and target images could actually have white background, I think that'll look better
- I'm not sure if the legend could ever obscure the actual plan, any way to have a close/toggle button? Or a smaller button that will show it when hovering?

ThinkingBIGRon commented:
100% agree.  Pushed an updated version that:

* boxes for will/will not execute
* moved to top left
* updated arrow backgrounds
* adds toggle and close buttons on the whitebox (tried a few options and this felt the best)

Screenshot:

![OnPaste 20200616-205603](https://user-images.githubusercontent.com/590503/84839739-2aac7080-b014-11ea-963c-b9635f6cea90.png)

File generated with the updated with this PR:

[execution-plan.zip](https://github.com/nuke-build/nuke/files/4789641/execution-plan.zip)

matkoch commented:
Almost perfect! Could you make the target boxes to have a bit of margin and a gray border as the ones from the graph? Then it's almost ready to merge.. I would just update part of the text. We can also make the "This target executes" one of the default targets (which is already shown in yellow), and the "This target does not execute" one of the non-defaults.

ThinkingBIGRon commented:
@matkoch - updated with your notes.

Nuke's build script plan output post updates:

[execution-plan.zip](https://github.com/nuke-build/nuke/files/4794066/execution-plan.zip)
